### PR TITLE
My second refactor

### PR DESCRIPTION
**Pull request by aqsasaied**

### CODE REVIEW

**Comment by aqsasaied** [comment]
We know it's not very good but we want to fix our dropdowns before the demo

File: src/screens/select.tsx [hunk 1]
```diff
@@ -108,42 +98,61 @@ const SelectScreen: React.FC<SelectScreenProps> = ({ navigation }) => {
...
-  };
-  const incChildren = () => {
-    if (children < 6) {
-      setChildren(children + 1);
-    }
-  };
-  const decChildren = () => {
-    if (children > 0) {
-      setChildren(children - 1);
+  const showOrHidePicker = (dateOrTime: boolean, show: boolean) => {
```
Comment (biadars) [inline src/screens/select.tsx:101]: I have no idea what this is doing and it looks like you might want to rename some of it/the parameters to make that a bit clearer

File: src/screens/select.tsx [hunk 2]
```diff
@@ -108,42 +98,61 @@ const SelectScreen: React.FC<SelectScreenProps> = ({ navigation }) => {
...
+        headers: {
+          'X-API-KEY': config.apiKey,
+          Accept: 'application/json',
+          'Content-Type': 'application/json',
+        },
+      })
+        .then((response) => response.json())
+        .then((response) => response as Stations)
+        .then((response) => {
+          if (response.error) {
```
Comment (biadars) [inline src/screens/select.tsx:135]: It'd be great to pull this out into a method and pass it into your then

**Review by biadars — COMMENTED** [review]
Just two speedy comments so you can merge this in quick

File: src/screens/select.tsx (hunk 2 above)
Comment (aqsasaied) [reply src/screens/select.tsx:135]: that's how we broke things earlier, we will attempt again in the remaining 20mins

**Pull request merged**